StorageTests/TestSQLiteHistoryRecommendations.swift

```diff
@@ -189,6 +189,61 @@ class TestSQLiteHistoryRecommendations: XCTestCase {
         let highlights = history.getHighlights().value.successValue!
         XCTAssertEqual(highlights.count, 0)
     }
+
+    /*
+     * Verify that we return one highlight per domain
+     *
+     */
+    func testMultipleDomainInHighlights() {
+        let db = BrowserDB(filename: "browser.db", files: files)
+        let prefs = MockProfilePrefs()
+        let history = SQLiteHistory(db: db, prefs: prefs)
+
+        let startTime = NSDate.nowMicroseconds()
+        let oneHourAgo = startTime - oneHourInMicroseconds
+        let fifteenMinutesAgo = startTime - 15 * microsecondsPerMinute
+
+        /*
+         * Site A: 1 visit, 1 hour ago = highlight
+         * Site B: 1 visits, 15 minutes ago = non-highlight
+         * Site C: 3 visits, 1 hour ago = highlight with the same domain
+         * Site D: 4 visits, 1 hour ago = non-highlight
+         */
+        let siteA = Site(url: "http://www.facebook.com", title: "A")
+        let siteB = Site(url: "http://siteB/", title: "B")
+        let siteC = Site(url: "http://www.facebook.com/", title: "C")
+        let siteD = Site(url: "http://siteD/", title: "D")
+
+        let siteVisitA1 = SiteVisit(site: siteA, date: oneHourAgo, type: .Link)
+        let siteVisitB1 = SiteVisit(site: siteB, date: fifteenMinutesAgo, type: .Link)
+
+        let siteVisitC1 = SiteVisit(site: siteC, date: oneHourAgo, type: .Link)
+        let siteVisitC2 = SiteVisit(site: siteC, date: oneHourAgo + 1000, type: .Link)
+        let siteVisitC3 = SiteVisit(site: siteC, date: oneHourAgo + 2000, type: .Link)
+
+        let siteVisitD1 = SiteVisit(site: siteD, date: oneHourAgo, type: .Link)
+        let siteVisitD2 = SiteVisit(site: siteD, date: oneHourAgo + 1000, type: .Link)
+        let siteVisitD3 = SiteVisit(site: siteD, date: oneHourAgo + 2000, type: .Link)
+        let siteVisitD4 = SiteVisit(site: siteD, date: oneHourAgo + 3000, type: .Link)
+
+        history.clearHistory().succeeded()
+        history.addLocalVisit(siteVisitA1).succeeded()
+
+        history.addLocalVisit(siteVisitB1).succeeded()
+
+        history.addLocalVisit(siteVisitC1).succeeded()
+        history.addLocalVisit(siteVisitC2).succeeded()
+        history.addLocalVisit(siteVisitC3).succeeded()
+
+        history.addLocalVisit(siteVisitD1).succeeded()
+        history.addLocalVisit(siteVisitD2).succeeded()
+        history.addLocalVisit(siteVisitD3).succeeded()
+        history.addLocalVisit(siteVisitD4).succeeded()
+
+        let highlights = history.getHighlights().value.successValue!
+        print(highlights.count)
+        XCTAssertEqual(highlights.count, 1)
```

Comment:
Nit: Also, since this test case just tests removing duplicate highlights, I would clean this up a bit and don't bother adding non-facebook.com sites just to make the case more clear.